src/main/scala/mesosphere/marathon/api/HttpModule.scala

```diff
@@ -0,0 +1,193 @@
+package mesosphere.marathon
+package api
+
+import com.typesafe.scalalogging.StrictLogging
+import java.io.File
+
+import org.eclipse.jetty.http.HttpVersion
+import org.eclipse.jetty.security._
+import org.eclipse.jetty.security.authentication.BasicAuthenticator
+import org.eclipse.jetty.server._
+import org.eclipse.jetty.server.handler.gzip.GzipHandler
+import org.eclipse.jetty.server.handler.{ HandlerCollection, RequestLogHandler }
+import org.eclipse.jetty.servlet.{ ServletContextHandler }
+import org.eclipse.jetty.util.security.{ Constraint, Password }
+import org.eclipse.jetty.util.ssl.SslContextFactory
+import org.rogach.scallop.ScallopOption
+
+/**
+  * Module which initializes the Jetty instance, and various handlers, given an httpConf
+  *
+  * @param conf The configuration used to initialize Jetty
+  */
+class HttpModule(conf: HttpConf) extends StrictLogging {
+  lazy val marathonHttpService: MarathonHttpService = new MarathonHttpService(httpServer)
+  lazy val requestLog: RequestLog = new JettyRequestLog()
+  lazy val httpServer: Server = {
+
+    val server = new Server()
+
+    val httpConfig = new HttpConfiguration()
+    httpConfig.setSecureScheme("https")
+    httpConfig.setSecurePort(conf.httpsPort())
+    httpConfig.setOutputBufferSize(32768)
+    httpConfig.setSendServerVersion(false)
+
+    def addConnector(name: String)(connector: Option[Connector]): Unit = {
+      connector match {
+        case Some(conn) =>
+          logger.info(s"Adding $name support.")
+          server.addConnector(conn)
+        case _ =>
+          logger.info(s"No $name support configured.")
+      }
+    }
+
+    val httpConnector: Option[Connector] = getHTTPConnector(server, httpConfig)
+    addConnector("HTTP")(httpConnector)
+
+    val httpsConnector: Option[Connector] = getHTTPSConnector(server, httpConfig)
+    addConnector("HTTPS")(httpsConnector)
+
+    // verify connector configuration
+    (httpConnector, httpsConnector) match {
+      case (Some(_), Some(_)) =>
+        logger.warn("Both HTTP and HTTPS support have been configured. " +
+          s"Consider disabling HTTP with --${conf.disableHttp.name}")
+      case (None, None) =>
+        throw new IllegalArgumentException(
+          "Invalid configuration: Neither HTTP nor HTTPS support has been configured.")
+      case _ =>
+      // everything seems fine
```

Comment:
Nit: Move on same line with `case _ =>` or indent.

Reply:
Weird code formatter didn't do it. Will do.

Reply:
✅